packages/core/src/awsService/cloudWatchLogs/commands/tailLogGroup.ts

```diff
@@ -3,17 +3,154 @@
     const wizard = new TailLogGroupWizard(logData)
     const wizardResponse = await wizard.run()
     if (!wizardResponse) {
         throw new CancellationError('user')
     }
 
-    //TODO: Remove Log. For testing while we aren't yet consuming the wizardResponse.
-    getLogger().info(JSON.stringify(wizardResponse))
+    const liveTailSessionConfig: LiveTailSessionConfiguration = {
+        logGroupName: wizardResponse.regionLogGroupSubmenuResponse.data,
+        logStreamFilter: wizardResponse.logStreamFilter,
+        logEventFilterPattern: wizardResponse.filterPattern,
+        region: wizardResponse.regionLogGroupSubmenuResponse.region,
+    }
+    const session = new LiveTailSession(liveTailSessionConfig)
+    if (registry.has(session.uri)) {
+        await prepareDocument(session)
+        return
+    }
+    registry.set(session.uri, session)
+
+    const document = await prepareDocument(session)
+    registerTabChangeCallback(session, registry, document)
+    const stream = await session.startLiveTailSession()
+
+    await handleSessionStream(stream, document, session)
+}
+
+export function closeSession(sessionUri: vscode.Uri, registry: LiveTailSessionRegistry) {
+    const session = registry.get(sessionUri)
+    if (session === undefined) {
+        throw new ToolkitError(`No LiveTail session found for URI: ${sessionUri.toString()}`)
+    }
+    session.stopLiveTailSession()
+    registry.delete(sessionUri)
+}
+
+export async function clearDocument(textDocument: vscode.TextDocument) {
+    const edit = new vscode.WorkspaceEdit()
+    const startPosition = new vscode.Position(0, 0)
+    const endPosition = new vscode.Position(textDocument.lineCount, 0)
+    edit.delete(textDocument.uri, new vscode.Range(startPosition, endPosition))
+    await vscode.workspace.applyEdit(edit)
+}
+
+async function prepareDocument(session: LiveTailSession): Promise<vscode.TextDocument> {
+    const textDocument = await vscode.workspace.openTextDocument(session.uri)
+    await clearDocument(textDocument)
+    await vscode.window.showTextDocument(textDocument, { preview: false })
+    await vscode.languages.setTextDocumentLanguage(textDocument, 'log')
+    return textDocument
+}
+
+async function handleSessionStream(
+    stream: AsyncIterable<StartLiveTailResponseStream>,
+    document: vscode.TextDocument,
+    session: LiveTailSession
+) {
+    try {
+        for await (const event of stream) {
+            if (event.sessionUpdate !== undefined && event.sessionUpdate.sessionResults !== undefined) {
+                const formattedLogEvents = event.sessionUpdate.sessionResults.map<string>((logEvent) =>
+                    formatLogEvent(logEvent)
+                )
+                if (formattedLogEvents.length !== 0) {
+                    await updateTextDocumentWithNewLogEvents(formattedLogEvents, document, session.maxLines)
+                }
+            }
+        }
+    } catch (err) {
+        throw new ToolkitError('Caught on-stream exception')
```

Comment:
what is the point of this? the  exception that was caught likely has more useful information.

please review *all* of the code guidelines: https://github.com/aws/aws-toolkit-vscode/blob/master/docs/CODE_GUIDELINES.md#exceptions

Reply:
ack, will adress.